contracts/ethereum/SnapshotXZodiacModule/ProposalRelayer.sol

```diff
@@ -15,64 +15,73 @@ contract SnapshotXProposalRelayer is Guardable {
    * @param _starknetCore Address of the StarkNet Core contract
-   * @param _decisionExecutorL2 Address of the new decision executor contract
+   * @param _l2ExecutionRelayer Address of the new decision executor contract
    */
-  function setUpSnapshotXProposalRelayer(address _starknetCore, uint256 _decisionExecutorL2)
+  function setUpSnapshotXProposalRelayer(address _starknetCore, uint256 _l2ExecutionRelayer)
     internal
   {
     starknetCore = IStarknetCore(_starknetCore);
-    decisionExecutorL2 = _decisionExecutorL2;
+    l2ExecutionRelayer = _l2ExecutionRelayer;
   }
 
   /**
    * @dev Changes the StarkNet decision executor contract
-   * @param _decisionExecutorL2 Address of the new decision executor contract
+   * @param _l2ExecutionRelayer Address of the new execution relayer contract
    */
-  function changeDecisionExecutorL2(uint256 _decisionExecutorL2) public onlyOwner {
-    decisionExecutorL2 = _decisionExecutorL2;
-    emit ChangedDecisionExecutorL2(_decisionExecutorL2);
+  function changeL2ExecutionRelayer(uint256 _l2ExecutionRelayer) public onlyOwner {
+    l2ExecutionRelayer = _l2ExecutionRelayer;
+    emit ChangedL2ExecutionRelayer(_l2ExecutionRelayer);
   }
 
   /**
    * @dev Receives L2 -> L1 message containing proposal execution details
-   * @param executionDetails Hash of all the transactions in the proposal
+   * @param executionHashLow Lowest 128 bits of the hash of all the transactions in the proposal
+   * @param executionHashHigh Highest 128 bits of the hash of all the transactions in the proposal
    * @param hasPassed Whether the proposal passed
    */
-  function _recieveFinalizedProposal(uint256 executionDetails, uint256 hasPassed) internal {
-    uint256[] memory payload = new uint256[](2);
-    payload[0] = executionDetails;
+  function _receiveFinalizedProposal(
+    uint256 callerAddress,
+    uint256 hasPassed,
+    uint256 executionHashLow,
+    uint256 executionHashHigh
+  ) internal {
+    uint256[] memory payload = new uint256[](4);
+    payload[0] = callerAddress;
     payload[1] = hasPassed;
+    payload[2] = executionHashLow;
+    payload[3] = executionHashHigh;
+
     /// Returns the message Hash. If proposal execution message did not exist/not received yet, then this will fail
-    starknetCore.consumeMessageFromL2(decisionExecutorL2, payload);
+    starknetCore.consumeMessageFromL2(l2ExecutionRelayer, payload);
   }
 
   /**
    * @dev Checks whether proposal has been received on L1 yet
-   * @param executionDetails Hash of all the transactions in the proposal
+   * @param executionHash Hash of all the transactions in the proposal
    * @param hasPassed Whether the proposal passed
    * @return isReceived Has the proposal been received
    */
-  function isFinalizedProposalRecieved(uint256 executionDetails, uint256 hasPassed)
+  function isFinalizedProposalReceived(uint256 executionHash, uint256 hasPassed)
```

Comment:
This needs to be updated to the new message payload format. Tbh we dont rly need this function right? I just added it for the sake of it but we could remove

Reply:
removed in 6e560643a1cfc0d511668f7c15e049449c9f8118 :D